sherpa/astro/ui/tests/test_astro_ui_plot.py

```diff
@@ -3579,3 +3579,163 @@ def test_set_plot_opt_explicit_astro():
     assert fig.axes[3].get_yscale() == 'linear'
 
     plt.close(fig)
+
+
+@requires_pylab
+@pytest.mark.parametrize("idval", [None, "bob"])
+def test_plot_fit_resid_handles_data_log(idval, clean_astro_ui):
+    """Check that log handling is correct: data=log
+
+    See also test_plot_fit_resid_handles_resid_log.
+
+    I thought we had tests of this, but apparently not.
+    Copy of test_ui_plot version but for PHA data.
+    """
+
+    from matplotlib import pyplot as plt
+
+    setup_example(idval)
+    ui.set_xlog('data')
+    ui.plot_fit_resid(idval)
+
+    fig = plt.gcf()
+    axes = fig.axes
+    assert len(axes) == 2
+    assert axes[0].xaxis.get_label().get_text() == ''
+
+    assert axes[0].xaxis.get_scale() == 'log'
+    assert axes[0].yaxis.get_scale() == 'linear'
+
+    assert axes[1].xaxis.get_scale() == 'log'
+    assert axes[1].yaxis.get_scale() == 'linear'
+
+
+@requires_pylab
+@pytest.mark.parametrize("idval", [None, "bob"])
+def test_plot_fit_resid_handles_resid_log(idval, clean_astro_ui):
+    """Check that log handling is correct: resid=log
+
+    We need to decide whether we want the residual setting to override
+    the linear display of the fit plot here. At present the code is
+    that if resid has xlog set then both will be drawn logged (since
+    the X axis is shared via a sharex=True argument to plt.subplots)
+    but we may decide this should change.
+
+    """
+
+    from matplotlib import pyplot as plt
+
+    setup_example(idval)
+    ui.set_xlog('resid')
+    ui.plot_fit_resid(idval)
+
+    fig = plt.gcf()
+    axes = fig.axes
+    assert len(axes) == 2
+    assert axes[0].xaxis.get_label().get_text() == ''
+
+    assert axes[0].xaxis.get_scale() == 'log'
+    assert axes[0].yaxis.get_scale() == 'linear'
+
+    assert axes[1].xaxis.get_scale() == 'log'
+    assert axes[1].yaxis.get_scale() == 'linear'
+
+
+@requires_pylab
+@pytest.mark.parametrize("idval", [None, "bob"])
+def test_plot_bkg_fit_resid_handles_data_log(idval, clean_ui):
+    """Check that log handling is correct: data=log"""
```

Comment:
The code below actually tests for data=linear

Reply:
No. It checks what happens when you've set the data "option/name/class" to log - i.e. set_xlog('data'). It just so happens that this setting isn't used in plot_bkg_fit_xxx (or, rather, I just want code to check what the actual behavior is, as this is one of those cases where I can argue for the x axis being linear and log).

Reply:
Although this is now outdated with the refactoring you suggested below.